File: csharp/test/JsonTag.Test/TestTypes.cs
namespace JsonTag.Test;

[JsonTag]
public abstract record Shape;

public sealed record Circle : Shape
{
    public double Radius { get; init; }
}

public sealed record Rectangle : Shape
{
    public double Width { get; init; }
    public double Height { get; init; }
}

[JsonTag]
public interface IVehicle;

[JsonTagName("Car")]
public sealed record PersonalVehicle : IVehicle
{
    public string Make { get; init; } = "";
}

[JsonTagName("Truck")]
public sealed record TransportationVehicle : IVehicle
{
    public double PayloadTons { get; init; }
}

File: fsharp/test/JsonTag.FSharp.Test/TestTypes.fs
namespace JsonTag.FSharp.Test

open System.Text.Json.Serialization

type Shape =
    | Circle of radius: float
    | Rectangle of width: float * height: float

type Vehicle =
    | [<JsonName("Car")>] PersonalVehicle of make: string
    | [<JsonName("Truck")>] TransportationVehicle of payloadTons: float

